### PR TITLE
AzureMonitor: Remove `simplejson`

### LINKED ISSUE #61882: Remove simplejson from AzureMonitor
The `simplejson` library is [no longer recommended by the backend team](https://github.com/grafana/grafana/blob/main/contribute/style-guides/backend.md?plain=1#L189). Therefore the  @grafana/partner-plugins  team has decided to factor the library out of our plugins in favor of the `encoding/json` package that's part of the Go standard library.

- Previous work for Cloudwatch https://github.com/grafana/grafana/pull/51062

**Implementation guidelines**
- Do not use anonymous structs
- Existing tests should pass unless they rely on `simplejson` as well

### PR DESCRIPTION
**Pull request by ZakharE**
**What is this feature?**
Remove usage of `simplejson` package and use built-in go `json` package instead

**Which issue(s) does this PR fix?**:
Fixes #61882

### CODE REVIEW

File: pkg/tsdb/azuremonitor/azuremonitor.go [hunk 1]
```diff
@@ -99,7 +94,13 @@ func NewInstanceSettings(cfg *setting.Cfg, clientProvider *httpclient.Provider,
 			return nil, fmt.Errorf("error reading settings: %w", err)
 		}
 
-		cloud, err := getAzureCloud(cfg, jsonData)
+		var jsonData types.AzureClientSettings
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/azuremonitor.go:97]: Could I suggest rather than doing multiple unmarshals, we create a `struct` composed of both `AzureClientSettings` and `AzureMonitorSettings` and then we will only have to unmarshal twice rather than 3 times 😊

File: pkg/tsdb/azuremonitor/credentials_test.go [hunk 2]
```diff
@@ -170,26 +170,22 @@ func TestCredentials_getAzureCredentials(t *testing.T) {
...
-			}
+			cfg := &setting.Cfg{}
 
 			credentials, err := getAzureCredentials(cfg, jsonData, secureJsonData)
 			require.NoError(t, err)
 			require.IsType(t, &azcredentials.AzureClientSecretCredentials{}, credentials)
 			clientSecretCredentials := credentials.(*azcredentials.AzureClientSecretCredentials)
 
-			assert.Equal(t, azsettings.AzureChina, clientSecretCredentials.AzureCloud)
+			assert.Equal(t, azsettings.AzureUSGovernment, clientSecretCredentials.AzureCloud)
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/credentials_test.go:188]: Why the change in the configuration here?

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go [hunk 3]
```diff
@@ -139,15 +142,15 @@ func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, logger
 		dataResponse.Frames = frames
 		return dataResponse
 	}
-
-	model, err := simplejson.NewJson(query.JSON)
+	var model reqAzureBodySubscriptions
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:145]: The `dataquery.AzureMonitorQuery` type can be used here instead of defining a new type.

**Review by aangelisc — CHANGES_REQUESTED** [review]
Thank you for this contribution @ZakharE! I've added some requests for changes, let me know what you think 😊

File: pkg/tsdb/azuremonitor/azuremonitor.go (hunk 1 above)
Comment (ZakharE) [reply pkg/tsdb/azuremonitor/azuremonitor.go:97]: Thanks! Will fix it

File: pkg/tsdb/azuremonitor/macros/macros.go [hunk 4]
```diff
@@ -111,14 +115,13 @@ func (m *kqlMacroEngine) evaluateMacro(logger log.Logger, name string, defaultTi
...
+			var model queryInterval
+			err := json.Unmarshal(m.query.JSON, &model)
 			if err != nil {
 				logger.Warn("Unable to parse model from query", "JSON", m.query.JSON)
 				it = defaultInterval
 			} else {
-				it, err = interval.GetIntervalFrom(&datasources.DataSource{
-					JsonData: simplejson.NewFromAny(dsInfo.JSONData),
-				}, model, defaultInterval)
+				it, err = intervalv2.GetIntervalFrom("", model.Interval, model.IntervalMs, defaultInterval)
```
[the remark this answers is not in the document]
Comment (ZakharE) [reply pkg/tsdb/azuremonitor/macros/macros.go:124]: Sorry about that 😞  
What do you think if i do it like this? 
`				var (
					dsInterval string
					ok         bool
				)
				if dsInterval, ok = dsInfo.JSONData["interval"].(string); !ok {
					dsInterval = ""
				}
				it, err = intervalv2.GetIntervalFrom(dsInterval, queryInterval.Interval, queryInterval.IntervalMs, defaultInterval)
`

File: pkg/tsdb/azuremonitor/credentials_test.go (hunk 2 above)
Comment (ZakharE) [reply pkg/tsdb/azuremonitor/credentials_test.go:188]: In previous version we set up test like this 
```
	jsonData := simplejson.NewFromAny(map[string]interface{}{
			"azureAuthType": azcredentials.AzureAuthClientSecret,
			"cloudName":     azUSGovManagement,
			"tenantId":      "9b9d90ee-a5cc-49c2-b97e-0d1b0f086b5c",
			"clientId":      "849ccbb0-92eb-4226-b228-ef391abd8fe6",
		})
```
`azUSGovManagement` is not a string. It is route defined in `routes.go`. Then in test we add following configuration 
```
			cfg := &setting.Cfg{
				Azure: &azsettings.AzureSettings{
					Cloud: azsettings.AzureChina,
				},
			}
```
And because `azUSGovManagement` is not a string we took cloud name from `cfg.Azure.Cloud` 
So i decide to  change config that cloudName in test setup will be equal in assertion. Which is  AzureUSGovernment

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go (hunk 3 above)
Comment (ZakharE) [reply pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:145]: Will fix. Thank you!

**Comment by ZakharE** [comment]
Hi @aangelisc! Thank you for review. I've pushed some changes 
Please, have a look when you will have time

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go [hunk 5]
```diff
@@ -139,15 +138,15 @@ func (e *AzureResourceGraphDatasource) executeQuery(ctx context.Context, logger
...
+	var model dataquery.AzureMonitorQuery
+	err := json.Unmarshal(query.JSON, &model)
 	if err != nil {
 		dataResponse.Error = err
 		return dataResponse
 	}
 
 	reqBody, err := json.Marshal(map[string]interface{}{
-		"subscriptions": model.Get("subscriptions").MustStringArray(),
+		"subscriptions": model,
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:149]:
```suggestion
		"subscriptions": model.Subscriptions,
```

**Review by aangelisc — APPROVED** [review]
Just one more nit and this is good to go! Thank you for the contribution 😊 You also have the following lint errors:
```
pkg/tsdb/azuremonitor/macros/macros.go:6: File is not `goimports`-ed (goimports)
	"github.com/grafana/grafana/pkg/tsdb/intervalv2"
pkg/tsdb/azuremonitor/credentials.go:4: File is not `goimports`-ed (goimports)
	"fmt"
pkg/tsdb/azuremonitor/credentials_test.go:4: File is not `goimports`-ed (goimports)
	"github.com/grafana/grafana/pkg/tsdb/azuremonitor/types"
```

**Comment by ZakharE** [comment]
Thank you for review! 

If there are any free issues in this package, I would be happy to help

**Comment by aangelisc** [comment]
Thanks @ZakharE! If you could just sync main as well we should be good to merge 😊

**Review by aangelisc — APPROVED** [review]
